doc/contexts-that-disable-exit-on-error.md

```diff
@@ -0,0 +1,87 @@
+Contexts That Disable Exit On Error
+===================================
+
+In wick, everything by default runs in [strict mode](bash-strict-mode.md).  This, among other things, sets `-e` which tells bash to exit if a command that was ran exitted with an error code.  It will also exit with the same error code as the command that failed.
+
+    set -e
+
+    errorCommand() {
+        false
+    }
+
+    echo "Will see this"
+    errorCommand()
+    echo "Will not see this"
+
+This will exit with the code 1.  We sometimes use exit codes to our advantage.
+
+    set -e
+
+    errorCommand() {
+        false
+    }
+
+    if ! errorCommand; then
+        echo "It returned a non-zero code, signally false or failure."
+        # continue to do other stuff.
+    fi
+
+Most of the time this works fine, but there are situations that might cause confusion.  What the `if` does in this case is disables the `-e` flag which is why it doesn't just exit at that line.
+
+Examples
+--------
+
+Here are a few examples that might cause confusion.
+
+    set -e
+
+    errorCommand() {
+        false
+        echo "shouldn't get here right?"
+    }
+
+    if ! errorCommand; then
+        echo "Expected failure"
+    else
+        echo "Didn't expect a success"
+    fi
+
+This outputs
+
+    shouldn't get here right?
+    Didn't expect a success
+
+Two things to note:
+1. Without `-e` `errorCommand` continues even though the first command exited with a non-zero value (`false`).
```

Comment:
5d51e5c1f3c4ef53e7d251dbb06c9830c1ad3c11

As a note, did you want this newline because it is the standard or because it is easier to read?  I believe it is displaying correctly.